src/analysis/inner_product_space/finite_dimensional.lean

```diff
@@ -0,0 +1,354 @@
+/-
+Copyright (c) 2023 Monica Omar. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Monica Omar
+-/
+import linear_algebra.invariant_submodule
+import analysis.inner_product_space.adjoint
+import analysis.inner_product_space.spectrum
+
+/-!
+# Finite-dimensional inner product spaces
+
+In this file, we prove some results in finite-dimensional inner product spaces.
+
+## Notation
+
+This file uses the local notation `P _` for `orthogonal_projection _`
+and `↥P _` for the extended orthogonal projection `V →L[ℂ] V`.
+
+We let `V` be an inner product space over `ℂ`.
+-/
+
+variables {V : Type*} [inner_product_space ℂ V]
+
+local notation `P` := orthogonal_projection
+local notation `↥P` := orthogonal_projection.extend
+
+-- the extended orthogonal projection is an invariant subspace
+lemma submodule.invariant_of_ortho_proj (U : submodule ℂ V) [finite_dimensional ℂ V] :
+  submodule.invariant U (↥P U) := λ x hx, set_like.coe_mem (P U x : U)
```

Comment:
The docstring seems weird, and the name should just be `submodule.invariant_orthogonal_projection'`. Also, you should try replacing most of your `[finite_dimensional ℂ V]` assumption by `[complete_space V]` (or even `[complete_space U]`), it should be enough in most cases